client/me/purchases/manage-purchase/index.jsx

```diff
@@ -87,7 +87,7 @@ class ManagePurchase extends Component {
 		userId: PropTypes.number,
 	};
 
-	componentWillMount() {
+	UNSAFE_componentWillMount() {
 		if ( ! this.isDataValid() ) {
 			page.redirect( purchasesRoot );
 			return;
```

Comment:
Also works:

```
if ( ! this.isDataValid() ) {
			return page.redirect( purchasesRoot );
		
```

Reply:
I'm going to disagree here - `return page.redirect( purchasesRoot );` makes it look like the return value matters, and makes me wonder what `page.redirect()` returns and what React will do with it. Naturally `UNSAFE_componentWillMount()` doesn't care about the return value, but I had to go and look it up to be sure, whereas that's explicit there in the code with the separate `return;`